crates/stages/src/stages/execution.rs

```diff
@@ -138,8 +138,12 @@ impl<EF: ExecutorFactory> ExecutionStage<EF> {
         tx: &mut Transaction<'_, DB>,
         input: ExecInput,
     ) -> Result<ExecOutput, StageError> {
-        let start_block = input.checkpoint().block_number + 1;
-        let max_block = input.previous_stage_checkpoint_block_number();
+        let range = input.next_block_range();
+        if range.is_empty() {
+            return Ok(ExecOutput::done(input.checkpoint().with_block_number(*range.end())))
```

Comment:
we shouldn't use range end since it's the progress of the previous stage and they might be different. with that said we should also make execution stage respect the `Pipeline::max_block` parameter. currently it is respected only if the `start..=max_block` range count satisfies the commit conditions

Reply:
> they might be different

why, if we have an assumption that all stages reach the target block one by one and have the same block number at the end?

> with that said we should also make execution stage respect the `Pipeline::max_block` parameter. 

shouldn't we do it for all stages in the pipeline when we execute the stage?

Reply:
> we should also make execution stage respect the `Pipeline::max_block` parameter

https://github.com/paradigmxyz/reth/pull/3030

Reply:
Currently, only `TransactionLookup` and `SenderRecovery` stages return the previous progress of current stage (after your changes with tx-based commits). Other stages do `ExecOutput::done(input.previous_stage_checkpoint_block_number())`. I think it's a generic behaviour that needs to be the same across all stages, and your changes seem to be the correct way to do that.

Updated the PR.